docs/factories.rst

```diff
@@ -0,0 +1,242 @@
+=========
+Factories
+=========
+
+The first versions of ``wired`` had a simple factory registration protocol:
+
+- Provide a callable and a class/interface to register for
+- That callable gets passed the container and returns the class/interface "instance"
+
+This can be streamlined and expanded.
+
+Current Implementation
+======================
+
+Imagine a ``LoginService``:
+
+.. code-block:: python
+
+    class LoginService:
+        def __init__(self):
+            ...
+
+With ``wired``, you have a factory function that can get information from the container, construct an instance, and return it:
+
+.. code-block:: python
+
+    def login_factory(container):
+        return LoginService()
+
+In the existing ``wired``, registering a factory means is a method call on the registry:
+
+.. code-block:: python
+
+    registry.register_factory(login_factory, LoginService)
+
+Thus, the first parameter is the factory function and the second is the "type" it is registering to handle.
+This is a simplified example:
+
+- ``register_factory`` doesn't *have* to take a second argument -- it can default to ``Interface`` which basically means "anything".
+
+- The ``register_factory`` method also exists on ``ServiceContainer`` but that's a detail.
+
+For such a simple case, the ``login_factory`` function isn't really needed.
+In fact, several pieces can be streamlined, especially in the presence of decorator support and type hinting.
+Let's take a look.
+
+The Simplest Case
+=================
+
+The above -- contrived -- example just constructs a class.
+Meaning, the class *constructor* is entirety of the factory.
+Imagine eliminating the ``def login_factory`` function and just doing this directly:
+
+.. code-block:: python
+
+    registry.register_service(LoginService)
```

Comment:
This is taking things a step too far and coupling the interface/context/name into a "service descriptor" which is a combination of `(factory, interface, context, name)`. I thought the plan was just to focus on the factory part of that and making it easier to define.